lib/active_shipping/external_return_label_request.rb

```diff
@@ -0,0 +1,479 @@
+# -*- coding: utf-8 -*-
+module ActiveShipping
+
+  class ExternalReturnLabelRequest
+
+    CAP_STRING_LEN = 100
+
+    USPS_EMAIL_REGEX = /^([a-zA-Z0-9_\-\.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([a-zA-Z0-9\-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$/
+
+    LABEL_FORMAT = {
+      'Instructions' => 'null',
+      'No Instructions' => 'NOI',
+      'Double Label' => 'TWO'
+    }
+
+    SERVICE_TYPE_CODE = [
+      '044', '019', '596', '020', '597','022', '024', '017', '018'
+    ]
+
+    CALL_CENTER_OR_SELF_SERVICE = ['CallCenter', 'Customer']
+
+    LABEL_DEFINITION = ['4X6', 'Zebra-4X6', '4X4', '3X6']
+
+    IMAGE_TYPE = ['PDF', 'TIF']
+
+    attr_reader :customer_name,
+                :customer_address1,
+                :customer_address2,
+                :customer_city,
+                :customer_state,
+                :customer_zipcode,
+                :customer_urbanization,
+                :company_name,
+                :attention,
+                :label_format,
+                :label_definition,
+                :service_type_code,
+                :merchandise_description,
+                :insurance_amount,
+                :address_override_notification,
+                :packaging_information,
+                :packaging_information2,
+                :call_center_or_self_service,
+                :image_type,
+                :address_validation,
+                :sender_name,
+                :sender_email,
+                :recipient_name,
+                :recipient_email,
+                :recipient_bcc,
+                :merchant_account_id,
+                :mid
+
+    def initialize(options = {})
+      options.each do |pair|
+        # TODO: Do we need to handle non-existent options
```

Comment:
Can we avoid `TODO`s please? They are pretty well hidden and quickly become outdated.

Fix the problem, make an issue, or raise an error.